### ISSUE TITLE
Using 100% CPU and Unresponsive on Arch - OpenSSL related

### ISSUE THREAD

**Issue by jeffska** [emesene/emesene#1099]
Using latest Emesene in Arch Linux with OpenSSL 1.0.1-1 uses 100% CPU and is generally unresponsive.  Downgrading to OpenSSL 1.0.0 solves the issue.

See:
https://groups.google.com/group/emesene/browse_thread/thread/3ef09221790e7f51/5ce7e0251ed601cf#5ce7e0251ed601cf
https://bbs.archlinux.org/viewtopic.php?id=138074

**Comment by c10ud** [comment]
Not an emesene issue then

2012/3/27 jeffska
reply@reply.github.com:

**Comment by lfottaviano** [comment]
I have archlinux and I havent seen this issue before

**Comment by jeffska** [comment]
Apparently OpenSSL 1.0.1 has changed some portion of negotiation which causes timeouts and other problems on some servers. Could be non standard SSL implementations or something.  Based on some things I found, I forced emesene to use SSL3 in OpenSSL by setting its context:

```
context = OpenSSL.Context(OpenSSL.SSLv3_METHOD)
```

on line 51 of emesene/e3/papylib/papyon/papyon/gnet/io/ssl_socket.py.  Works much better now.  CPU usage and application responsiveness are back to normal. My guess is that the MSN servers are some of those that aren't playing nice with the new SSL/TLS negotiation.
